cli/lib/tasks/process-js-task.js

```diff
@@ -0,0 +1,381 @@
+const appc = require('node-appc');
+const { IncrementalFileTask } = require('appc-tasks');
+const crypto = require('crypto');
+const fs = require('fs-extra');
+const jsanalyze = require('node-titanium-sdk/lib/jsanalyze');
+const path = require('path');
+const pLimit = require('p-limit');
+const { promisify } = require('util');
+
+const i18n = appc.i18n(__dirname);
+const __ = i18n.__;
+const limit = pLimit(8);
+
+/**
+ * Task that processes JS files by applying several transforms and copying them to their
+ * destination inside the build directory.
+ */
+class ProcessJsTask extends IncrementalFileTask {
+	/**
+	 * Constructs a new processing task.
+	 *
+	 * @param {Object} options Configuration object for this task.
+	 * @param {String} [options.name='process-js'] Name for the task.
+	 * @param {String[]} options.inputFiles Array of input files this task will use.
+	 * @param {String[]} options.incrementalDirectory Path to a folder where incremental task data will be stored.
+	 * @param {Object} options.logger The logger instance used by this task.
+	 * @param {Object} options.builder iOS builder instance.
+	 * @param {String[]} options.jsFiles Array with info about JS files to resolve paths.
+	 * @param {String[]} options.jsBootstrapFiles Array of bootstrap scripts to consider. The task will directly modify this array.
+	 * @param {String} options.sdkCommonFolder Path to common JS files from the SDK.
+	 * @param {Object} options.defaultAnalyzeOptions Default configuration options for jsanalyze.
+	 * @param {Boolean} options.defaultAnalyzeOptions.minify Wether to minify the JS files or not.
+	 * @param {Boolean} options.defaultAnalyzeOptions.transpile Wether to transpile the JS files or not.
+	 * @param {Boolean} options.defaultAnalyzeOptions.sourceMaps Wether to generate source maps or not.
+	 * @param {String} options.defaultAnalyzeOptions.resourcesDir .
```

Comment:
This is used in node-titanium-sdk to know where to insert the polyfills, but is no longer used in master/8.1.0